zetaclient/chains/base/observer.go

```diff
@@ -377,7 +400,82 @@ func (ob *Observer) ReadLastBlockScannedFromDB() (uint64, error) {
 	return lastBlock.Num, nil
 }
 
-// EnvVarLatestBlock returns the environment variable for the latest block by chain.
+// LoadLastTxScanned loads last scanned tx from environment variable or from database.
+// The last scanned tx is the tx hash from which the observer should continue scanning.
+func (ob *Observer) LoadLastTxScanned(logger zerolog.Logger) {
+	// get environment variable
+	envvar := EnvVarLatestTxByChain(ob.chain)
+	scanFromTx := os.Getenv(envvar)
+
+	// load from environment variable if set
+	if scanFromTx != "" {
+		logger.Info().
```

Comment:
Can't we use logger in ob object?

Reply:
https://github.com/zeta-chain/node/commit/eb814a91d89bee5be48174081af7338f76cb197a